ui/nuxeo-routing-behavior.js

```diff
@@ -161,7 +179,7 @@ export const RoutingBehavior = {
     if (!routeVal) {
       throw new Error(`invalid router key: ${routeKey}`);
     }
-    return fn(routeVal, ...params);
+    return { path: fn(routeVal, ...params), baseUrl };
```

Comment:
I thought the plan was to have this just return a URL as before, adding the base URL to it when needed.
Then we could just skip the `generateUrl` call when we have a full URI

Reply:
@nelsonsilva what you suggesting is to call the generateUrl instead here once, but only when we need have a full URL (i.e., when we are dealing with more than one repository)?

Reply:
But in practice this is already what's happening, right? I just moved that part of the magic from the `_routeEntity` to the `urlFor` and `nagivateTo`. And I did it because there are two types of relative URLs: 1) with hashbang, as consumed by urlFor, and without (consumed by navigateTo with pagejs). This private method returns the `baseUrl` as a separate property so that the `urlFor` and the `navigateTo` can build the relative urls as they need it.

Reply:
yes, the end result would be the same. It's really a matter of responsibility here as i feel it's up to the routing functions (wether they're explicit named routes or implicit by entity type) to build and return the final URL.
This would allow for instance supporting a use case where my `document` reverse routing function returned an absolute URL for managed documents whereas here you don't support that use case since the baseUrl is not controlled by my custom function.
Anyway, I understand there are implementation details that make it simpler to have separated path and baseUrl and this is definitely not a blocker so feel free to merge as is 👍